File: Everything/Models/Lifting/LiftSetLink.cs
using System.Collections.Generic;

namespace everything.Models
{
    public class LiftSetLink
    {
        public int Id { get; set; }
        public string Notes { get; set; }

        public int LiftId { get; set; }
        public virtual Lift Lift { get; set; }

        public int? LiftingWorkoutId { get; set; }
        public virtual LiftingWorkout LiftingWorkout { get; set; }

        public IEnumerable<LiftSet> LiftSets { get; set; }
    }
}

File: Everything/ClientApp/src/models/lifting/LiftSetLink.ts
import Lift from "./Lift";
import LiftSet from "./LiftSet";

export default class LiftSetLink {
    id: number = 0;
    notes: string | null = null;
    sets: LiftSet[] = [];
    lift: Lift | null = null;

    constructor(init?: Partial<LiftSetLink>) {
        Object.assign(this, init);
    }
}